ui/app/src/components/PublishingStatusButton/PublishingStatusButtonUI.tsx

```diff
@@ -69,14 +75,15 @@ export const PublishingStatusButtonUI = forwardRef<HTMLButtonElement, Publishing
         {/* TODO:
             The spinner might be better suited to be on the PublishingStatusAvatar component
             so when we have progress, it is show everywhere the publishing avatar shows up. */}
-        {(isFetching || status === 'publishing') && (
+        {(isFetching || isInProgressPublishingStatus(status)) && (
           <CircularProgress
             size={
               // Default progress size matches small button, but the medium
               // size (which is this component's default) needs a larger spinner
               ['medium', void 0].includes(rest.size) ? 48 : void 0
             }
-            variant="indeterminate"
+            value={Math.round((numberOfItems / totalItems) * 100)}
+            variant={isInProgressPublishingStatus(status) ? 'determinate' : 'indeterminate'}
```

Comment:
We still have the same problem from previous set of change requests. We're running the `[].includes()` twice. The idea was to running it once.